lib/elixir/test/elixir/kernel/with_test.exs

```diff
@@ -40,6 +40,10 @@ defmodule Kernel.WithTest do
     assert result == :error
   end
 
+  test "else conditions" do
+    assert with({:ok, res} <- 41, do: res, else: ({:error, error} -> error; res -> res + 1)) == 42
```

Comment:
It has something with catch all condition:
 
```
res -> res + 1
```

where we add another catch all condition for raising of error and get warning, because of
2 catch all conditions in a case.

I've added check, if we have catch all condition in elses, in that case we do not add raising catch all clause.